keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1368 @@
+    // the plugin supports the feature. Boolean features have
+    // no parameters, merely listing them indicates support.
+    Parameters runtime.RawExtension
+}
+
+type ResourceClaim struct {
+    // The plugin must set a finalizer here before it attempts to
+    // allocate the resource. It removes the finalizer again when
+    // a) the allocation attempt has definitely failed or b) when
+    // the allocated resource was freed. This ensures that
+    // resources are not leaked.
+    ObjectMeta
+    // Spec describes the desired attributes of a resource that then
+    // needs to be allocated. It can only be set once when creating
+    // the ResourceClaim.
+    Spec ResourceClaimSpec
+    // Status describes whether the resource is available and with
+    // which attributes.
+    Status ResourceClaimStatus
+}
+
+type ResourceClaimSpec struct {
+    // ResourceClassName references the plugin and additional
+    // parameters via the name of a ResourceClass that was
+    // created as part of the plugin deployment.
+    ResourceClassName string
+
+    // Parameters holds arbitrary values that will be available to the plugin
+    // when allocating a resource for the claim.
+    Parameters runtime.RawExtension
+
+    // Allocation can start immediately or when a Pod wants to use
+    // the resource. Waiting for a Pod is the default.
+    AllocationMode AllocationMode
+}
+
+type AllocationMode string
+
+const (
+     AllocationModeImmediate AllocationMode = “Immediate”
+     AllocationModeDelayed AllocationMode = “Delayed”
+}
+
+type ResourceClaimStatus struct {
+   // Explains what the current status of the claim is and
+   // determines which component needs to do something.
+   Phase ResourceClaimPhase
+
+   // When allocation is delayed, the scheduler must set
+   // the node for which it wants the resource to be allocated
+   // before the plugin proceeds with allocation.
+   // For immediate allocation, the scheduler will not set
+   // this field. The plugin controller component may then
+   // set it to trigger allocation on a specific node if the
+   // resources are local to nodes. 
+   SelectedNode string
+
+   // When allocation is delayed, and the scheduler needs to
+   // decide on which node a Pod should run, it must first
+   // ask the plugin on which nodes the resource might be
+   // made available. To trigger that check, the scheduler must
+   // provide the names of nodes which might be suitable
+   // for the Pod.
+   PotentialNodes []string
+
+   // A change of the node candidates triggers a check
+   // on which nodes the resource could be made available.
+   // This can change, so the plugin must refresh
+   // this information periodically until a node gets
+   // selected by the scheduler.
+   SuitableNodes []string
+
+   // An allocated resource is available on nodes that match this
+   // selector. If nil, the resource is available everywhere.
+   Topology *corev1.NodeSelector
```

Comment:
Agreed, a better term would be useful. It came from CSI.

I don't like `AvailableNodes` because its not the nodes which are available. `AvailableOnNodes` is more accurate, but longer.

Reply:
but OTOH if CSI uses `Topology` for the exact same purpose, keeping the same field name might also be justified. my thinking was to harmonize PotentialNodes/SuitableNodes/Available(On)Nodes.

Reply:
The topology concept in CSI is more complex, for example it involves mapping key/value pairs provided by the driver onto Kubernetes nodes. We can and should use something simpler here - we just need a good name.

Reply:
+1

Reply:
So did we agree on `AvailableOnNodes`?

Reply:
`AvailableOnNodes` sounds good to me